File: VB/PieChartOtherSliceExample/Form1.vb
Imports DevExpress.DashboardCommon
Imports DevExpress.DashboardCommon.ViewerData
Imports DevExpress.DashboardWin
Imports DevExpress.XtraEditors
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq

Namespace PieChartOtherSliceExample
    Partial Public Class Form1
        Inherits XtraForm

        ' Stores the selection state. 
        ' The parameter modification resets the selection. This variable is used to restore it.
        Private selectionState As New List(Of AxisPoint)()
        Public Sub New()
            InitializeComponent()
            AddHandler dashboardViewer1.ConfigureDataConnection, AddressOf dashboardViewer1_ConfigureDataConnection

            AddHandler dashboardViewer1.DashboardItemVisualInteractivity, AddressOf dashboardViewer1_DashboardItemVisualInteractivity
            AddHandler dashboardViewer1.DashboardItemSelectionChanged, AddressOf dashboardViewer1_DashboardItemSelectionChanged

            dashboardViewer1.LoadDashboard("Dashboard.xml")
        End Sub

        Private Sub dashboardViewer1_DashboardItemVisualInteractivity(ByVal sender As Object, ByVal e As DashboardItemVisualInteractivityEventArgs)
            If e.DashboardItemName = "gridDashboardItem1" Then
                e.SelectionMode = DashboardSelectionMode.Multiple
                e.SetDefaultSelection(selectionState)
            End If
        End Sub
        Private Sub dashboardViewer1_DashboardItemSelectionChanged(ByVal sender As Object, ByVal e As DashboardItemSelectionChangedEventArgs)
            Dim viewer As DashboardViewer = DirectCast(sender, DashboardViewer)
            If e.DashboardItemName = "gridDashboardItem1" Then
                selectionState = e.CurrentSelection.Select(Function(tuple) tuple.GetAxisPoint(DashboardDataAxisNames.DefaultAxis)).ToList()
                Dim stringSelection As IEnumerable(Of String) = selectionState.Select(Function(p) p.Value).Cast(Of String)()
                viewer.Parameters("ParamSalesPerson").SelectedValues = stringSelection
            End If
        End Sub
        Private Sub dashboardViewer1_ConfigureDataConnection(ByVal sender As Object, ByVal e As DashboardConfigureDataConnectionEventArgs)
            Dim parameters As ExtractDataSourceConnectionParameters = TryCast(e.ConnectionParameters, ExtractDataSourceConnectionParameters)
            If parameters IsNot Nothing Then
                parameters.FileName = Path.GetFileName(parameters.FileName)
            End If
        End Sub
    End Class
End Namespace

File: CS/PieChartOtherSliceExample/Form1.cs
using DevExpress.DashboardCommon;
using DevExpress.DashboardCommon.ViewerData;
using DevExpress.DashboardWin;
using DevExpress.XtraEditors;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace PieChartOtherSliceExample
{
    public partial class Form1 : XtraForm
    {
        // Stores the selection state. 
        // The parameter modification resets the selection. This variable is used to restore it.
        List<AxisPoint> selectionState = new List<AxisPoint>();
        public Form1()
        {
            InitializeComponent();
            dashboardViewer1.ConfigureDataConnection += dashboardViewer1_ConfigureDataConnection;

            dashboardViewer1.DashboardItemVisualInteractivity += dashboardViewer1_DashboardItemVisualInteractivity;
            dashboardViewer1.DashboardItemSelectionChanged += dashboardViewer1_DashboardItemSelectionChanged;

            dashboardViewer1.LoadDashboard("Dashboard.xml");
        }

        private void dashboardViewer1_DashboardItemVisualInteractivity(object sender, DashboardItemVisualInteractivityEventArgs e) {
            if(e.DashboardItemName == "gridDashboardItem1") {
                e.SelectionMode = DashboardSelectionMode.Multiple;
                e.SetDefaultSelection(selectionState);
            }
        }  
        private void dashboardViewer1_DashboardItemSelectionChanged(object sender, DashboardItemSelectionChangedEventArgs e) {
            DashboardViewer viewer = (DashboardViewer)sender;
            if(e.DashboardItemName == "gridDashboardItem1") {
                selectionState = e.CurrentSelection.Select(tuple => tuple.GetAxisPoint(DashboardDataAxisNames.DefaultAxis)).ToList();
                IEnumerable<string> stringSelection = selectionState.Select(p => p.Value).Cast<string>();
                viewer.Parameters["ParamSalesPerson"].SelectedValues = stringSelection;
            }
        }
        private void dashboardViewer1_ConfigureDataConnection(object sender, DashboardConfigureDataConnectionEventArgs e)
        {
            ExtractDataSourceConnectionParameters parameters = e.ConnectionParameters as ExtractDataSourceConnectionParameters;
            if (parameters != null)
                parameters.FileName = Path.GetFileName(parameters.FileName);
        }
    }
}
